### PR TITLE
Documentation: Correcting typos and wording in Concepts section

### PR DESCRIPTION
**Pull request by CasperWA**
As I read through the Concepts section of the docs, I fixed some typos, wordings, and performed minor restructuring, when I felt it was necessary.

### CODE REVIEW

File: docs/source/concepts/workflows.rst [hunk 1]
```diff
@@ -156,25 +156,25 @@ The resulting provenance would look like the following:
...
+    The provenance generated by the work chain example that computes the sum and product directly in its outline steps, instead of delegating it to calculation functions
 
 Note how, in contrast with the provenance of the previous correct solution from :numref:`fig_work_chains_provenance_add_multiply_workchain_full`, there are no explicit calculation nodes representing the computation of the sum and the product.
 Instead, all that computation is abstracted and represented by the single workflow node that represents the execution of the work chain.
 The logic inside of those outline steps is then 'hidden' or 'encapsulated' in the provenance graph by a single workflow node.
-Additionally, the output node representing the final product, only has a ``return`` link, even though it was 'created' by the work chain.
-This is because :ref:`workflow processes do not have the capacity to create new nodes<working_workfunctions_returning_data>`, and therefore in a sense in this example, the data provenance is lost.
+Additionally, the output node representing the final product, only has a ``return`` link, even though it was "*created*" by the work chain.
```
Comment (sphuber) [inline docs/source/concepts/workflows.rst]: before you changed to *created*, let's stick to that

File: docs/source/concepts/calculations.rst [hunk 2]
```diff
@@ -53,31 +53,33 @@ To solve this, one only has to wrap them in the :py:class:`~aiida.orm.nodes.data
...
 
 The only difference with the previous snippet is that all inputs have been wrapped in the :py:class:`~aiida.orm.nodes.data.int.Int` class.
-The result that is returned by the function, likewise, is a :py:class:`~aiida.orm.nodes.data.int.Int` node that contains the result of the computation.
+The result that is returned by the function, is now also an :py:class:`~aiida.orm.nodes.data.int.Int` node that can be stored in the provenance graph,
+and contains the result of the computation.
 
 .. note::
-	Since ``x`` and ``y`` inside the function are already :py:class:`~aiida.orm.nodes.data.int.Int` instances the sum will also be one, since all arithmetic operators are overloaded.
+	Since ``x`` and ``y`` inside the ``add`` and ``multiply`` functions are already :py:class:`~aiida.orm.nodes.data.int.Int` instances the sum will also be one.
+    This is true because all arithmetic operators also work on the base AiiDA classes (``Int``, ``Float``, etc.) as they would the equivalent python types.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/calculations.rst]: Done

File: docs/source/concepts/calculations.rst [hunk 3]
```diff
@@ -91,18 +93,19 @@ To interface an external code with the engine of AiiDA, the :py:class:`~aiida.en
 A detailed explanation of how to implement it, the interface and best practices, can be found in a :ref:`later section<working_calcjobs>`.
 Here, instead, we will focus on the big picture and explain in broad lines how a calculation job models the execution of an external code and what tasks it performs when launched.
 
-To illustrate how a calculation job operates, we need an example external code.
-Let's imagine an external code that consists of a bash script that reads an input file containing two integers, sums them and echoes the result, for example:
+To illustrate how a calculation job operates, we need an external code.
+Let's imagine an external code that consists of a bash script that reads an input file containing two integers, sums them and displays the result in the standard output environment (usually done by using ``echo`` in a terminal), for example:
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/calculations.rst]: Done

File: docs/source/concepts/calculations.rst [hunk 4]
```diff
@@ -91,18 +93,19 @@ To interface an external code with the engine of AiiDA, the :py:class:`~aiida.en
...
     #!/bin/bash
-    # Read two integers from file 'aiida.in` and echo their sum
+    # Read two integers from file 'aiida.in' and echo their sum
     x=$(cat aiida.in | awk '{print $1}')
     y=$(cat aiida.in | awk '{print $2}')
     echo $(( $x + $y ))
 
-When run, this script reads the content of a file called ``aiida.in`` and expects that it contains two integers, that it will parse into the variables ``x`` and ``y``, whose sum it will echo.
+When run, this script reads the contents of a file called ``aiida.in`` and expects that it contains two integers.
+It will then parse these into the variables ``x`` and ``y``, whose sum it will echo.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/calculations.rst]: done

File: docs/source/concepts/calculations.rst [hunk 5]
```diff
@@ -140,20 +144,17 @@ Transport tasks
 To arrive at the provenance graph shown above in :numref:`fig_calculation_jobs_provenance_arithmetic_add`, the engine performed quite some tasks.
 When a calculation job is launched, the engine will take it roughly through the following steps:
 
- * Upload: the calculation job implementation is used to transform the input nodes into the required input files, which are uploaded to a 'working' directory on the target machine
- * Submit: a job to execute the calculation is submitted to the scheduler of the computer on which the input `code` is configured.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/calculations.rst]: done

File: docs/source/concepts/processes.rst [hunk 6]
```diff
@@ -69,27 +69,28 @@ Waiting   Finished
...
 When a process is first created, it is put in the ``Created`` state.
 As soon as it is picked up by a runner and it is active, it will be in the ``Running`` state.
 If the process is waiting for another process, that it called, to be finished, it will be in the ``Waiting`` state.
-A process that is in the ``Killed`` state, means that the user issued a command to kill it, or its parent process was killed.
-The ``Excepted`` state indicates that during execution an exception occurred that was not caught and the process was unexpectedly terminated.
-The final option is the ``Finished`` state, which means that the process was successfully executed, and the execution was nominal.
-Note that this does not automatically mean that the result of the process can also considered to be successful, it just executed without any problems.
+If a process is in the ``Killed`` state, it means the user issued a command to kill it, or its parent process was killed.
+The ``Excepted`` state indicates that during execution an exception occurred that was not caught, and the process was unexpectedly terminated.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/processes.rst]: this is *obviously* left as a check that you were doing a concentrated and "woke" review ... (done)

File: docs/source/concepts/processes.rst [hunk 7]
```diff
@@ -69,27 +69,28 @@ Waiting   Finished
...
+The final option is the ``Finished`` state, which means that the process was successfully executed, and the excecution was nominal.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/processes.rst]: right! My bad

File: docs/source/concepts/processes.rst [hunk 8]
```diff
@@ -69,27 +69,28 @@ Waiting   Finished
...
+Note that this does not automatically mean that the result of the process can also be considered to be successful, it was just executed without any problems.
 
-To distinghuis between a successful and a failed execution, there is the :ref:`exit status<concepts_process_exit_codes>`.
+To distinguish between a successful and a failed execution, there is the :ref:`exit status<concepts_process_exit_codes>`.
 This is another attribute that is stored in the node of the process and is an integer that can be set by the process.
-A zero means that the result of the process was successful, and a non-zero value indicates a failure.
-All the process nodes used by the various processes are a sub class of :py:class:`~aiida.orm.nodes.process.ProcessNode`, which defines handy properties to query the process state and exit status.
+A ``0`` (zero) means that the result of the process was successful, and a non-zero value indicates a failure.
+All the process nodes used by the various processes are sub-classes of :py:class:`~aiida.orm.nodes.process.ProcessNode`,
+which defines handy properties to query the process state and exit status.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/processes.rst]: done

File: docs/source/concepts/processes.rst [hunk 9]
```diff
@@ -105,10 +106,10 @@ When you load a calculation node from the database, you can use these property m
 Process exit codes
 ==================
 The previous section about the process state showed that a process that is ``Finished`` does not say anything about whether the result is 'successful' or 'failed'.
-The ``Finished`` state means nothing more than that the engine managed to run the process to the end of execution without it encountering exceptions or being killed.
+The ``Finished`` state means nothing more than that the engine succeeded to run the process to the end of execution, without it encountering exceptions or being killed.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/processes.rst]: done

File: docs/source/concepts/processes.rst [hunk 10]
```diff
@@ -135,24 +136,26 @@ This means that the output of many of the ``verdi`` commands, such as ``verdi pr
 Process tasks
 -------------
 The previous section explained how launching a process means creating an instance of the ``Process`` class in memory.
-When the process is being 'ran' (see the section on :ref:`launching processes<working_processes_launch>` for more details), that is to say in a local interpreter, that process instance will die as soon as the interpreter dies.
-This is what makes 'submitting' a process, often times the preferred launching method.
+When the process is being 'ran' (see the section on :ref:`launching processes<working_processes_launch>` for more details) that is to say in a local interpreter,
+the particular process instance will die as soon as the interpreter dies.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/processes.rst]: done

File: docs/source/concepts/processes.rst [hunk 11]
```diff
@@ -135,24 +136,26 @@ This means that the output of many of the ``verdi`` commands, such as ``verdi pr
...
+This is what often makes 'submitting' a process the preferred method of launching a calculation or workflow.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/processes.rst]: done

File: docs/source/concepts/processes.rst [hunk 12]
```diff
@@ -135,24 +136,26 @@ This means that the output of many of the ``verdi`` commands, such as ``verdi pr
...
 When a process is 'submitted', an instance of the ``Process`` is created, along with the node that represents it in the database, and its state is then persisted to the database.
 This is called a 'process checkpoint', more information on which :ref:`will follow later<concepts_process_checkpoints>`.
 Subsequently, the process instance is shutdown and a 'continuation task' is sent to the process queue of RabbitMQ.
 This task is simply a small message that just contains an identifier for the process.
 
 All the daemon runners, when they are launched, subscribe to the process queue and RabbitMQ will distribute the continuation tasks to them as they come in, making sure that each task is only sent to one runner at a time.
-The receiving daemon runner can restore the process instance in memory from the checkpoint that was stored in the database and continue the execution.
-As soon as the process reaches a terminal state, the daemon worker will acknowledge to RabbitMQ that the task has been completed.
+The receiving daemon runnerr can restore the process instance in memory from the checkpoint that was stored in the database and continue the execution.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/processes.rst]: done :)

File: docs/source/concepts/processes.rst [hunk 13]
```diff
@@ -135,24 +136,26 @@ This means that the output of many of the ``verdi`` commands, such as ``verdi pr
...
+As soon as the process reaches a terminal state, the daemon runner will acknowledge to RabbitMQ that the task has been completed.
 Until the runner has confirmed that a task is completed, RabbitMQ will consider the task as incomplete.
-If a daemon runner is shutdown or dies before it got the chance to finish running a process, the task will automatically be requeued by RabbitMQ and sent to another.
-Together with the fact that all the tasks in the process queue are persisted to disk by RabbitMQ, guarantees that once a continuation task has been sent to RabbitMQ, it will at some point be finished, while allowing the machine to be shutdown.
+If a daemon runner is shutdown or dies before it got the chance to finish running a process, the task will automatically be requeued by RabbitMQ and sent to another daemon runner.
+Together with the fact that all the tasks in the process queue are persisted to disk by RabbitMQ, guarantees that once a continuation task has been sent to RabbitMQ,
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/processes.rst]: done

File: docs/source/concepts/processes.rst [hunk 14]
```diff
@@ -135,24 +136,26 @@ This means that the output of many of the ``verdi`` commands, such as ``verdi pr
...
+it will at some point be finished, while allowing the machine to be shutdown.
 
 Each daemon runner has a maximum number of tasks that it can run concurrently, which means that if there are more active tasks than available slots, some of the tasks will remain queued.
-Processes whose task is in the queue and not with any runner, though technically 'active' as it is not terminated, it is not actually being run at the moment.
-While a process is not actually being run, i.e. it is not in memory with a runner, one also cannot interact with it.
-Similarly, as soon as the task disappears, either because the process was intentionally terminated, or unintentionally due to a bug or problem, the process will never continue running again.
+Processes whose task is in the queue and not with any runner, though technically 'active' as it is not terminated, is not actually being run at the moment.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/processes.rst]: done. Changed to plural in the interposed sentence as well.

File: docs/source/concepts/provenance.rst [hunk 15]
```diff
@@ -63,19 +63,20 @@ Implementation
 Graph nodes
 -----------
 
-The **nodes** of the AiiDA provenance graph can be grouped into two main **types**: **process nodes** (``ProcessNode``), that represent the execution of calculations or workflows, and **data nodes** (``Data``), that represent pieces of data.
+The **nodes** of the AiiDA provenance graph can be grouped into two main **types**: **process nodes** (``ProcessNode``),
+that represent the execution of calculations or workflows, and **data nodes** (``Data``), that represent pieces of data.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/provenance.rst]: done

File: docs/source/concepts/provenance.rst [hunk 16]
```diff
@@ -63,19 +63,20 @@ Implementation
...
 
 In particular, **process nodes** are divided into two sub categories:
 
     - **calculation nodes** (``CalculationNode``): Represent code execution that creates new data. These are further subdivided in two subclasses:
 
-        - ``CalcJobNode``: Represents the execution of a calculation external to AiiDA, typically via a job batch scheduler (see the concept of :ref:`calculation jobs<concepts_calcjobs>`).
-        - ``CalcFunctionNode``: Represents the execution of a python function (see the concept of :ref:`calculation functions<concepts_calcfunctions>`).
+        - :py:class:`~aiida.orm.nodes.CalcJobNode`: Represents the execution of a calculation external to AiiDA, typically via a job batch scheduler (see the concept of :ref:`calculation jobs<concepts_calcjobs>`).
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/provenance.rst]: will update them all to make sure.

[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/provenance.rst]: done

File: docs/source/concepts/provenance.rst [hunk 17]
```diff
@@ -119,32 +121,36 @@ By allowing workflow processes to *return* data, it can also *return* data that
...
 
 
 Graph examples
 --------------
 
-With these basic definitions of AiiDA’s provenance graph in place, let’s take a look at some more interesting.
-Consider the sequence of computations that adds two number `x` and `y` and multiplies the result with a third number `z`.
-This sequence as represented in the provenance graph would look something like is shown in :numref:`fig_provenance_add_multiply_data`.
+With these basic definitions of AiiDA’s provenance graph in place, let’s take a look at some examples.
+Consider the sequence of computations that adds two numbers `x` and `y` together, and then multiplies the result with a third number `z`.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/provenance.rst]: done

File: docs/source/concepts/provenance.rst [hunk 18]
```diff
@@ -119,32 +121,36 @@ By allowing workflow processes to *return* data, it can also *return* data that
...
+    The same calculation `(x+y)*z` is performed using a workflow. Here the data nodes D\ :sub:`1`, D\ :sub:`2`, and D\ :sub:`3` are the inputs of the workflow W\ :sub:`1`, which *calls* calculation C\ :sub:`1` with inputs D\ :sub:`1` and D\ :sub:`2`.
+    It then *calls* calculation C\ :sub:`2`, using as inputs D\ :sub:`3` and D\ :sub:`4` (which was *created* by C\ :sub:`2`\).
+    Calculation C\ :sub:`2` *creates* data node D\ :sub:`5`, which is finally *returned* by workflow W\ :sub:`1`\.
 
-Notice that if we were to omit the workflow nodes and all its links from the provenance graph in :numref:`fig_provenance_add_multiply_full`, one would end up with the exact same graph as shown in :numref:`fig_provenance_add_multiply_data`.
\ No newline at end of file
+Notice that if we were to omit the workflow nodes and all its links from the provenance graph in :numref:`fig_provenance_add_multiply_full`,
+one would end up with the exact same graph as shown in :numref:`fig_provenance_add_multiply_data` (the **data provenance** graph).
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/provenance.rst]: done

File: docs/source/concepts/workflows.rst [hunk 19]
```diff
@@ -27,15 +27,15 @@ Work functions
...
+To explain the use of the ``@workfunction``, we will continue with the example of the :ref:`calculation functions<concepts_calcfunctions>`, so before continuing, read that section first.
+The example showed how the ``@calcfunction`` decorator can be used to create two functions that, for three given integers, computes the sum of the first two, which is then multiplied with the third, all the while keeping the provenance.
 Even though the calculation functions ensured that the provenance of the data was kept, the logic of *who* called these functions was not explicitly kept.
-From the provenance graph generated by the calculation functions, it is impossible to deduce of the functions where called straight after another in a single script, or whether first the ``add`` function was called and a long time later, the output was used as an input for the ``multiply`` call.
+From the provenance graph generated by the calculation functions, it is impossible to deduce if the functions were called straight after another in a single script, or whether first the ``add`` function was called and a long time later, the output was used as an input for the ``multiply`` function.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/workflows.rst]: sure, done

File: docs/source/concepts/workflows.rst [hunk 20]
```diff
@@ -126,11 +126,11 @@ If we were to reimplement our work function solution of the simple example probl
...
-The only thing you need to notice here is that it defines the *inputs* that the work chain takes, its logical *outline* and the *outputs* that it will produce.
+The only thing you need to notice here, is that it defines the *inputs* that the work chain takes, its logical *outline* and the *outputs* that it will produce.
 The steps of the outline are implemented as class methods of the work chain.
 The ``add`` step will add the first two integers by calling the ``add`` calculation function, and store the sum temporarily in the :ref:`context<working_workchains_context>`.
 The next step in the outline, ``multiply``, will take the sum stored in the context that was computed in the first outline step and call the ``multiply`` calculation function with the third input integer.
-Finally, the ``result`` step will take the product produced by the previous step and record it as an output of the work chain.
+Finally, the ``result`` step will take the product produced by the previous step, and record it as an output of the work chain.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/workflows.rst]: done

File: docs/source/concepts/workflows.rst (hunk 1 above)
Comment (CasperWA) [reply docs/source/concepts/workflows.rst]: For this particular one, I just didn't want people to, in any shape or form, think that a work chain can *create* - but maybe it looks better overall without the quotation marks, and the readers will be able to understand just fine?

File: docs/source/concepts/calculations.rst [hunk 21]
```diff
@@ -53,31 +53,33 @@ To solve this, one only has to wrap them in the :py:class:`~aiida.orm.nodes.data
     :code: python
 
 The only difference with the previous snippet is that all inputs have been wrapped in the :py:class:`~aiida.orm.nodes.data.int.Int` class.
-The result that is returned by the function, likewise, is a :py:class:`~aiida.orm.nodes.data.int.Int` node that contains the result of the computation.
+The result that is returned by the function, is now also an :py:class:`~aiida.orm.nodes.data.int.Int` node that can be stored in the provenance graph,
+and contains the result of the computation.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/concepts/calculations.rst]: done - to my best knowledge.

File: docs/source/concepts/workflows.rst (hunk 1 above)
Comment (CasperWA) [reply docs/source/concepts/workflows.rst]: I have removed the quotations.

**Comment by sphuber** [comment]
Docs are failing yo @CasperWA if you fix them everything is good to go and I will approve and merge

**Comment by CasperWA** [comment]
> Docs are failing yo @CasperWA if you fix them everything is good to go and I will approve and merge

All's well that ends well. Only jenkins not working now - np.

**Pull request merged**